xrpl/models/transactions/credential_create.py

```diff
@@ -0,0 +1,84 @@
+"""Model for CredentialCreate transaction type."""
+
+from dataclasses import dataclass, field
+from typing import Dict, Optional
+
+from typing_extensions import Self
+
+from xrpl.models.required import REQUIRED
+from xrpl.models.transactions.transaction import Transaction
+from xrpl.models.transactions.types import TransactionType
+from xrpl.models.utils import HEX_REGEX, KW_ONLY_DATACLASS, require_kwargs_on_init
+
+
+@require_kwargs_on_init
+@dataclass(frozen=True, **KW_ONLY_DATACLASS)
+class CredentialCreate(Transaction):
+    """This transaction creates a Credential object. It must be sent by the issuer."""
+
+    transaction_type: TransactionType = field(
+        default=TransactionType.CREDENTIAL_CREATE,
+        init=False,
+    )
+
+    account: str = REQUIRED  # type: ignore
+    """
+    The issuer of the credential.
+    """
+    subject: str = REQUIRED  # type: ignore
+    """
+    The subject of the credential.
+    """
+
+    credential_type: str = REQUIRED  # type: ignore
+    """
+    A (hex-encoded) value to identify the type of credential from the issuer.
+    """
+    expiration: Optional[int] = None
+    """
+    Optional credential expiration.
+    """
+    uri: Optional[str] = None
+    """
+    Optional additional data about the credential (such as a link to the Verifiable
+    Credential document).
+    """
+
+    # Note: Validity checks pertaining to expiration, account or subject will need
+    # access to the last closed ledger (or) the latest state of the blockchain. Hence,
+    # these checks are omitted from this implementation.
+    def _get_errors(self: Self) -> Dict[str, str]:
+        return {
+            key: value
+            for key, value in {
+                **super()._get_errors(),
+                "uri": self._get_uri_error(),
+                "credential_type": self._get_credential_type_error(),
+            }.items()
+            if value is not None
+        }
+
+    def _get_uri_error(self: Self) -> Optional[str]:
+        if self.uri is not None:
+            error = ""
+            if len(self.uri) == 0:
+                error += "Length of URI field must be greater than 0. "
+            if len(self.uri) > 256:
```

Comment:
I recommend using the `_MAX_URI_LENGTH` variable here.

Reply:
fixed in d83f0d70